### ISSUE TITLE
[Feature] Use `igdb` tag to set the `IGDB ID` from filename

### ISSUE THREAD

**Issue by taki-sahli** [rommapp/romm#1333]
**Is your feature request related to a problem? Please describe.**
Sometimes, RomM mismatch the game with the correct one in IGDB.

**Describe the solution you'd like**
To avoid mismatch of games with `IGDB`, we can use special tag `igdb` and set it from the filename.

**Describe alternatives you've considered**

**Additional context**
For example this game:
- The correct match should:
![391319603-00c77ef4-4d7f-4059-9264-2d0f7338769b](https://github.com/user-attachments/assets/fe0b69ac-93e9-4ee7-bb23-ea7696e0c457)

- But the result returned from IGDB gives this:
![391319534-ae16c487-e962-42e6-ae35-da1d239a9fd1](https://github.com/user-attachments/assets/3300e65a-f47a-40d6-b3de-82b82c7b5b91)

- With this filename `Inside [rev-10,igdb-7342].7z`, there's no ambiguity.

**Comment by InsertDisc** [comment]
Inside [rev-10] [igdb-7342].7z
This would would be nice, yes.

**Comment by gantoine** [comment]
We're happy to support this, but we're not going to start changing filenames to append `igdb-****` tags since it's not a common standard. So unless you want to go through and manually tag your files with their IGDB IDs I don't see how this would be a particularly useful feature.

**Comment by InsertDisc** [comment]
I think OP meant (correct me if I'm wrong) that if the tag was there then try to match using that instead of the filename. So yes, they'd be manually added by server admins.

**Comment by taki-sahli** [comment]
Exactly:
  if igdb-tag exists: use it directly
  else normal behavior.
The main purpose of this tag, is the ability to fix ambiguity in the regular matcher.

**Comment by taki-sahli** [comment]
An idea:
To ease setting it, when an admin uses manual match, at this moment, RomM can automatically add the IGDB ID tag.
![image](https://github.com/user-attachments/assets/58bbe3ef-2506-4718-b942-79df8df4bb1a)

**Comment by gantoine** [comment]
>  RomM can automatically add the IGDB ID tag

We won't do that unless the community comes to a consensus that having the IGDB ID in the filename is good practice, like TVDB ID has been adopted by sonarr/radarr/jellyfin.

**Comment by InsertDisc** [comment]
I don't need the tag added by RomM, but the matching would be nice. I guess this is where OP and I differ.

**Comment by taki-sahli** [comment]
I see, using the tag only will be enough for this use case.

**Comment by omenariita** [comment]
It would be nice to implement a manual search of metadata by index, for unambiguous matching, in addition to the existing search by name. For example, entering "igdb:242599" I get one specific game. Perhaps I did not figure it out, and this has already been done, but so far I have not found any mention of this

**Comment by InsertDisc** [comment]
> It would be nice to implement a manual search of metadata by index, for unambiguous matching, in addition to the existing search by name. For example, entering "igdb:242599" I get one specific game. Perhaps I did not figure it out, and this has already been done, but so far I have not found any mention of this

Next to the text box, where it says name, you can change that to id and search igdb IDs.

**Comment by ilikepie3326** [comment]
Agreed, this would be super useful, and could be done using the same ID matching tag convention that Plex uses.
example: Game Title {tgdb:123456}.exe
